renderlib/ViewerWindow.cpp

```diff
@@ -56,17 +50,27 @@ ViewerWindow::setSize(int width, int height)
 }
 
 void
-ViewerWindow::showAreaLightGizmo(bool show)
+ViewerWindow::select(SceneObject* obj)
 {
-  // check if m_areaLightTool is in m_tools
-  bool isShowing = std::find(m_tools.begin(), m_tools.end(), m_areaLightTool) != m_tools.end();
-  if (isShowing == show) {
+  // if obj is currently selected then do nothing?
+  if (sceneView.selection == obj) {
     return;
   }
-  if (show) {
-    m_tools.push_back(m_areaLightTool);
-  } else {
-    m_tools.erase(std::remove(m_tools.begin(), m_tools.end(), m_areaLightTool), m_tools.end());
+
+  // deselect the current selection
+  if (sceneView.selection) {
+    ManipulationTool* tool = sceneView.selection->getSelectedTool();
+    if (tool) {
+      m_tools.erase(std::remove(m_tools.begin(), m_tools.end(), tool), m_tools.end());
```

Comment:
Can you explain what this line is doing? I see that it's basically unchanged from the diff but I wasn't sure how the `remove` and `erase` are working.

Reply:
Yeah this is a bit funky but a well established pattern.  The std::remove actually shifts the queried object (`tool`) to the end of the vector `m_tools` and returns an iterator to the start of the removed items.  Then erase actually drops everything from where that iterator starts.  
see here:
https://stackoverflow.com/questions/347441/how-can-you-erase-elements-from-a-vector-while-iterating

Reply:
Gotcha, thank you!